File: 03Builder/Students/mkacprzak/Builder/src/main/java/FluentBuilder/Computer.java
package FluentBuilder;

import java.util.BitSet;

public class Computer {

    private Computer(){}

    @Override
    public String toString() {
        return "Computer{" +
                "type=" + type +
                ", gpu=" + gpu +
                ", cpu=" + cpu +
                ", ram=" + ram +
                ", motherboard=" + motherboard +
                '}';
    }

    private Type type;
    private GPU gpu;
    private CPU cpu;
    private Ram ram;
    private Motherboard motherboard;

    public static Builder builder()
    {
        return new Builder();
    }


    public static class Builder {

        Computer computer = new Computer();

        private Builder()
        {}

        public Builder type(Type type)
        {
            computer.type=type;
            return this;
        }


        public Builder gpu(GPU gpu)
        {
            computer.gpu=gpu;
            return this;
        }
        public Builder cpu(CPU cpu)
        {
            computer.cpu=cpu;
            return this;
        }
        public Builder ram(Ram ram)
        {
            computer.ram=ram;
            return this;
        }

        public Builder motherboard(Motherboard mb)
        {
            computer.motherboard=mb;
            return this;
        }
        public Computer build()
        {
            return computer;
        }
    }



}

File: ProjektyStudentow/2018_Okoniewski_Daniel/lib/computer.rb
require_relative 'cpu'
require_relative 'drive'
require_relative 'gpu'
require_relative 'ram'
require_relative 'command'
require_relative 'memento'

# This is a facade of complex computer object
class Computer
  attr_accessor :ram, :drive, :cpu, :gpu, :model_name, :state, :id
  include ComputerCommands

  def execute(command)
    command.execute(self)
  end

  def build
    ComputerBuilder.instance.build
  end
end
